### PR TITLE
snap-seccomp, osutil: use osutil.AtomicFile in snap-seccomp

### PR DESCRIPTION
**Pull request by mvo5**
This PR adds a new osutil.AtomicFile.BackingFile() function that exposes the underlying file. This is then used in snap-seccomp to write the generated bpf in an atomic way.

It also switches the default for tests to snapdUnsafeIO=true and ensures the tests for the AtomicWriteTestSuite are run with both unsafe and safe IO.

I can split this PR up if you think that makes the review easier.

### CODE REVIEW

File: cmd/snap-seccomp/main.go [hunk 1]
```diff
@@ -790,27 +789,16 @@ func compile(content []byte, out string) error {
...
 	defer fout.Close()
-	if err := secFilter.ExportBPF(fout); err != nil {
-		return err
-	}
-	if err := fout.Sync(); err != nil {
-		return err
-	}
-	if err := os.Rename(out+".tmp", out); err != nil {
+
+	if err := secFilter.ExportBPF(fout.BackingFile()); err != nil {
```
Comment (niemeyer) [inline cmd/snap-seccomp/main.go]: The new method doesn't seem to be doing much good here. Why not using fout directly?

Comment (niemeyer) [reply cmd/snap-seccomp/main.go]: Not that fout itself (or fout.File) _are_ the backing file.

Comment (niemeyer) [reply cmd/snap-seccomp/main.go]: Note that fout itself (and fout.File) _are_ the backing file.

Comment (mvo5) [reply cmd/snap-seccomp/main.go]: The way osutil.NewAtomicFile() is written currently does not allow me to use it directly (or fout.File). The reason is that NewAtomicFile returns on interface of AtomicWriter not osutil.atomicFile. However that can be easily changed of course.

Details with the current code:
```
with just fout:
./main.go:798: cannot use fout (type osutil.AtomicWriter) as type *os.File in argument to secFilter.ExportBPF
with fout.File:
./main.go:798: fout.File undefined (type osutil.AtomicWriter has no field or method File)
```

I can of course tweak osutil/io.go and expose the osutil.AtomicFile itself, something like http://paste.ubuntu.com/25525999/ (plus updating all the tests).

**Comment by mvo5** [comment]
@chipaca Thanks a bunch for your feedback. I updated the code to follow my pastebin proposal. I think I added appropriate documentation but I would value feedback on this.

**Comment by chipaca** [comment]
Is good. I'd flip-flopped about it being an interface or not originally.

File: osutil/export_test.go [hunk 2]
```diff
@@ -93,13 +93,13 @@ func MockChown(f func(*os.File, int, int) error) func() {
...
-func SetAtomicFileRenamed(aw AtomicWriter, renamed bool) {
+func SetAtomicFileRenamed(aw *AtomicFile, renamed bool) {
 	println("renamed", renamed)
-	aw.(*atomicFile).renamed = renamed
+	aw.renamed = renamed
 }
 
-func GetAtomicFile(aw AtomicWriter) *os.File {
-	return aw.(*atomicFile).File
+func GetAtomicFile(aw AtomicFile) *os.File {
```
Comment (pedronis) [inline osutil/export_test.go]: we shouldn't need this one at all now, no?

**Pull request merged**